test/replication-luatest/gh_5295_split_brain_test.lua

```diff
@@ -140,26 +149,80 @@ g.test_async_old_term = function(cg)
     reconnect_and_check_split_brain(cg)
 end
 
--- Any unseen sync transaction confirmation from an obsolete term means a
+-- A conflicting sync transaction confirmation from an obsolete term means a
 -- split-brain.
-g.test_confirm_old_term = function(cg)
+g.test_bad_confirm_old_term = function(cg)
     partition_replica(cg)
     cg.split_replica:exec(write_promote)
     cg.main:exec(function() box.space.sync:replace{1} end)
     reconnect_and_check_split_brain(cg)
 end
 
--- Any unseen sync transaction rollback from an obsolete term means a
+-- Obsolete sync transaction confirmation might be fine when it doesn't
+-- contradict local history.
+g.test_good_confirm_old_term = function(cg)
+    t.tarantool.skip_if_not_debug()
+    -- Delay confirmation on the old leader, so that the transaction is included
+    -- into new leader's PROMOTE, but the old leader writes a CONFIRM for it.
+    cg.main:exec(function()
+        local lsn = box.info.lsn
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 1)
+        require('fiber').new(function() box.space.sync:replace{1} end)
+        t.helpers.retrying({}, function()
+            t.assert_equals(box.info.lsn, lsn + 1)
+        end)
+    end)
+    partition_replica(cg)
+    cg.split_replica:exec(write_promote)
+    cg.main:exec(function()
+        local lsn = box.info.lsn
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        t.helpers.retrying({}, function()
+            t.assert_equals(box.info.lsn, lsn + 1)
+        end)
+    end)
+    reconnect_and_check_no_split_brain(cg)
+end
+
+-- A conflicting sync transaction rollback from an obsolete term means a
 -- split-brain.
-g.test_rollback_old_term = function(cg)
+g.test_bad_rollback_old_term = function(cg)
+    t.tarantool.skip_if_not_debug()
+    -- Delay rollback on the old leader, so that the transaction is included
+    -- into new leader's PROMOTE, but the old leader writes a ROLLBACK for it.
```

Comment:
`included into leader's promote` -> `rolled back by leader's PROMOTE`.

Reply:
No, it's committed by the new leader, but rolled back by the old leader.

Reply:
Resolving, since you've approved the PR.